### PR TITLE
Prepare models for auto-update: match format, remove JSON comment

### PR DESCRIPTION
**Pull request by dagood**
Sorting the manifest and including a revision number in the tags matches it up with what auto-update generates, so the diff of the first auto-update will be purer. The JSON comment needed to go because Go's JSON parser fails when it sees it. The `buildArgs` object will be described in the manifest.json generator now--just an implementation detail of generating a file that the .NET Docker build infra can use.

I also added a description of the non-standard `version.json` keys to `src/microsoft/README.md`.

### CODE REVIEW

**Comment by dagood** [comment]
@microsoft/golang-compiler PTAL, this is some prep work for the repo to receive auto-updates from our builds.

**Review by jaredpar — APPROVED** [review]

File: src/microsoft/1.15/stretch/Dockerfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 
 ENV PATH /usr/local/go/bin:$PATH
 
-ENV GOLANG_VERSION 1.15.5
+ENV GOLANG_VERSION 1.15.14
```
Comment (jcouv) [inline src/microsoft/1.15/stretch/Dockerfile:20]: Are the docker files generated? For some reason GitHub collapses them from diff view by default.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

File: src/microsoft/1.15/stretch/Dockerfile (hunk 1 above)
Comment (dagood) [reply src/microsoft/1.15/stretch/Dockerfile:20]: Yep! This was set up by upstream in `.gitattributes`:

https://github.com/microsoft/go-docker/blob/b3b99a9fe654cfaadd4adf749717a49e0fb9c74b/.gitattributes#L1-L2

**Pull request merged**